libs/core/src/lib/slider/slider.component.ts

```diff
@@ -0,0 +1,748 @@
+            this._setRangeHandleValueAndPosition(RangeHandles.First, _position[0]);
+            this._setRangeHandleValueAndPosition(RangeHandles.Second, _position[1]);
+        }
+
+        this._value = value;
+        this._position = _position;
+
+        this.onChange(value);
+        this.onTouched();
+
+        this._cdr.markForCheck();
+    }
+
+    /** @hidden */
+    get _popoverValueRef(): number[] {
+        return [this._position as number, this._handle1Value, this._handle2Value];
+    }
+
+    /** @hidden */
+    @ViewChild('track', { read: ElementRef })
+    trackEl: ElementRef<HTMLDivElement>;
+
+    /** @hidden */
+    @ViewChild('handle', { read: ElementRef })
+    handle: ElementRef<HTMLDivElement>;
+
+    /** @hidden */
+    @ViewChild('rangeHandle1', { read: ElementRef })
+    rangeHandle1: ElementRef<HTMLDivElement>;
+
+    /** @hidden */
+    @ViewChild('rangeHandle2', { read: ElementRef })
+    rangeHandle2: ElementRef<HTMLDivElement>;
+
+    /** @hidden */
+    @ViewChildren(PopoverComponent)
+    _popovers: QueryList<PopoverComponent>;
+
+    /** @hidden */
+    _value: number | SliderTickMark | SliderTickMark[] | number[] = 0;
+
+    /** @hidden */
+    _min = 0;
+
+    /** @hidden */
+    _max = 100;
+
+    /** @hidden */
+    _step = 1;
+
+    /** @hidden */
+    _progress = 0;
+
+    /** @hidden */
+    _isRange = false;
+
+    /** @hidden */
+    _handle1Position = 0;
+
+    /** @hidden */
+    _handle2Position = 0;
+
+    /** @hidden */
+    _handle1Value = 0;
+
+    /** @hidden */
+    _handle2Value = 0;
+
+    /** @hidden */
+    _rangeProgress = 0;
+
+    /** @hidden */
+    _tickMarks: SliderTickMark[] = [];
+
+    /** @hidden */
+    _valuesBySteps: number[] = [];
+
+    /** @hidden */
+    _sliderValueTargets = SliderValueTargets;
+
+    /** @hidden */
+    _popoverInputFieldClass = `fd-slider-popover-input-${sliderId}`;
+
+    /** @hidden */
+    _isRtl = false;
+
+    /**
+     * @hidden
+     * An RxJS Subject that will kill the data stream upon component’s destruction (for unsubscribing)
+     */
+    private readonly _onDestroy$: Subject<void> = new Subject<void>();
+
+    /** @hidden */
+    constructor(
+        private readonly _elementRef: ElementRef,
+        private readonly _cdr: ChangeDetectorRef,
+        private readonly _renderer: Renderer2,
+        @Optional() private readonly _rtlService: RtlService
+    ) {}
+
+    /** @hidden */
+    ngOnInit(): void {
+        this._subscribeToRtl();
+        this.buildComponentCssClass();
+        this._checkIsInRangeMode();
+        this._attachResizeListener();
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this._constructTickMarks();
+        this._constructValuesBySteps();
+        this._onResize();
+    }
+
+    /** @hidden */
+    ngOnChanges(): void {
+        this.buildComponentCssClass();
+        this._checkIsInRangeMode();
+        this._constructTickMarks();
```

Comment:
`this._constructTickMarks();` calls `this.cd.detectChanges()` inside it.
It means that each `ngOnChanges` will lead to `cd.detectChanges()`, not sure that's a good idea...
Can we avoid it?